src/instr/instr_TI_trace.cpp

```diff
@@ -219,11 +236,49 @@ void print_TIPushState(paje_event_t event)
          ((pushState_t) event->data)->value->name);
     break;
   }
+ 
+  //We need to keep track of the current state, so we can register when the
+  //current iteration (TRACING_ITERATION) ends. 
+  // We will push the whole 'extra', because we need access to the printing
+  // function when the state is popped.
+  xbt_dynar_push(state_tracker, &extra);
+
+  xbt_free(process_id);
+  //'extra' will be freed in print_TIPopState
+}
+
+void print_TIPopState(paje_event_t event)
+{
+  char *process_id;
+  instr_extra_data extra;
+  FILE *trace_file;
+
+  //What is the current state?
+  xbt_dynar_pop(state_tracker, &extra);
+  
+  //This function is actually only usefull for tracing the iterations of the
+  //application.
+  if(extra->type != TRACING_CUSTOM){
+    return;
+  }
 
+  if (strstr(((pushState_t) event->data)->container->name, "rank-") == NULL)
+    process_id = xbt_strdup(((pushState_t) event->data)->container->name);
+  else
+    process_id = xbt_strdup(((pushState_t) event->data)->container->name + 5);
+
+  trace_file = (FILE *)xbt_dict_get(tracing_files,
+      ((pushState_t) event->data)->container->name);
+  
+  if(extra->print_pop)
+    extra->print_pop(trace_file, process_id, extra);
+  
   if (extra->recvcounts != nullptr)
     xbt_free(extra->recvcounts);
   if (extra->sendcounts != nullptr)
     xbt_free(extra->sendcounts);
-  xbt_free(process_id);
   xbt_free(extra);
+
+  return;
```

Comment:
What's the point of this 'return' at the end of a function? I see the point, actually, but that's not part of the SimGrid coding style.